Networking-in-Android-Lab/app/src/main/java/com/ezequielc/networking_in_android_lab/MainActivity.java

```diff
@@ -0,0 +1,125 @@
+
+    @Override
+    protected void onCreate(Bundle savedInstanceState) {
+        super.onCreate(savedInstanceState);
+        setContentView(R.layout.activity_main);
+
+        mCerealButton = (Button) findViewById(R.id.cereal_button);
+        mChocolateButton = (Button) findViewById(R.id.chocolate_button);
+        mTeaButton = (Button) findViewById(R.id.tea_button);
+        mRecyclerView = (RecyclerView) findViewById(R.id.recycler_view);
+
+        LinearLayoutManager linearLayoutManager =
+                new LinearLayoutManager(this, LinearLayoutManager.VERTICAL, false);
+        mRecyclerView.setLayoutManager(linearLayoutManager);
+
+        mProductsList = new ArrayList<>();
+        mRecyclerView.setAdapter(new ProductRVAdapter(mProductsList));
+
+        ConnectivityManager connectivityManager = (ConnectivityManager) getSystemService(CONNECTIVITY_SERVICE);
+        NetworkInfo info = connectivityManager.getActiveNetworkInfo();
+
+        if (info != null && info.isConnected()) {
+            Toast.makeText(this, "Connected to Network", Toast.LENGTH_SHORT).show();
+        } else {
+            Toast.makeText(this, "No Network Connection", Toast.LENGTH_SHORT).show();
+        }
+
+        View.OnClickListener onClickListener = new View.OnClickListener() {
+            @Override
+            public void onClick(View view) {
+                switch (view.getId()) {
+                    case R.id.cereal_button:
+                        performGetRequest(CEREAL_URL);
+                        break;
+                    case R.id.chocolate_button:
+                        performGetRequest(CHOCOLATE_URL);
+                        break;
+                    case R.id.tea_button:
+                        performGetRequest(TEA_URL);
+                        break;
+                }
+            }
+        };
+
+        mCerealButton.setOnClickListener(onClickListener);
+        mChocolateButton.setOnClickListener(onClickListener);
+        mTeaButton.setOnClickListener(onClickListener);
+    }
+
+    private void performGetRequest(String url){
+        OkHttpClient client = new OkHttpClient();
+
+        Request request = new Request.Builder()
+                .url(url)
+                .build();
+
+        client.newCall(request).enqueue(new Callback() {
+            @Override
+            public void onFailure(Call call, IOException e) {
+                e.printStackTrace();
+            }
+
+            @Override
+            public void onResponse(Call call, Response response) throws IOException {
+                if (!response.isSuccessful()) {
+                    throw new IOException("Unexpected code: " + response);
+                }
+
+                try {
+                    String responseBody = response.body().string();
+                    JSONObject results = new JSONObject(responseBody);
+                    JSONArray itemArr = results.getJSONArray("items");
+                    mProductsList.removeAll(mProductsList);
```

Comment:
It would be a little faster to just call `mProductsList.clear()` rather than `mProductsList.removeAll(mProductsList)` because the latter looks for each element of the list before trying to remove that element. `clear()` on the other hand, doesn't need to do any such comparisons, so it's quicker.